### ISSUE TITLE
ghc 8.2.1 (Cabal 2.x) can't compile stack  1.5.1

### ISSUE THREAD

**Issue by peti** [commercialhaskell/stack#3422]
I tried compiling the current Hackage version of stack with ghc-8.2.1, but the build failed with the following error in Setup.hs:

~~~
Setup.hs:7:31: error:
    Module
    ‘Distribution.Package’
    does not export
    ‘PackageName(PackageName)’
  |
7 | import Distribution.Package ( PackageName(PackageName), PackageId, InstalledPackageId, packageVersion, packageName )
  |                               ^^^^^^^^^^^^^^^^^^^^^^^^
~~~

Is this a known issue?

**Comment by phadej** [comment]
As far as I can see, 1.5.1 doesn't support GHC-8.2.1 (and doesn't advertise that). Do you use `--allow-newer` or something similar?

**Comment by peti** [comment]
I'm not quite sure what you mean by "supporting 8.2.1", exactly. I build tried to build the pristine stack 1.5.1 release from Hackage with ghc 8.2.1 within Nix, but cannot compile the Setup.hs file.

**Comment by phadej** [comment]
I see, indeed `setup-depends` doesn't have upper bound for `Cabal`.

It's a bit gray area. In theory, you can compile `stack` and its `Setup.hs` with different `Cabal`, e.g. with `cabal new-build` (note, it's GHC-8.0.2, and I'm using cabal-install HEAD)

```
% cabal new-configure --constraint='stack:setup.Cabal >= 2.0
```

But in practice neither `stack` or `cabal new-build` will do that.

With GHC-8.2.1 the `stack` itself won't build (and bounds restrict that), so there's no sense to build `./Setup.hs` - and neither cabal or stack will try that.

**Comment by lwm** [comment]
Will https://github.com/commercialhaskell/stack/pull/3421 resolve this?

**Comment by snoyberg** [comment]
Seems to be resolved by #3421, closing
